### PR TITLE
EOS-20858: Pacemaker Alert further log replication for >3nodes.

### PR DESCRIPTION
**Pull request by shindeam**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    EOS-20858: Pacemaker Alert further log replication for >3nodes.
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes
  </code>
</pre>
## Problem Description
<pre>
  <code>
    EOS-20858: Pacemaker Alert further log replication for >3nodes.
  </code>
</pre>
## Solution
<pre>
  <code>
    - Get all online corosync nodes
    - Get self nodeid
    - Find out way of identifying the highest node from online nodes
    - Send alert only from lowest or highest online node
    - Environment variable are available in self.crm_env
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
Logs of node2:
2021-07-23 06:05:13 alert_monitor INFO [__init__] AlertEventFilter initialized.
2021-07-23 06:05:13 alert_monitor INFO [process_alert] Processing event for NodeAlertMonitor
2021-07-23 06:05:13 alert_monitor INFO [get_node_id] List of online nodes ids in cluster in sorted ascending order: ['1', '2', '3']
2021-07-23 06:05:13 alert_monitor INFO [get_node_id] Local node name: srvnode-2.data.private
         Local node id: 2

2021-07-23 06:05:13 alert_monitor INFO [process_alert] Last Nodes IDs: 3
2021-07-23 06:05:13 alert_monitor INFO [process_alert] This node does not have highest id. Local node id : 2
, all nodes: ['1', '2', '3'].

Logs of node3:
2021-07-23 06:05:13 alert_monitor INFO [_get_online_nodes] List of online nodes ids in cluster in sorted ascending order: ['2', '3']
2021-07-23 06:05:13 alert_monitor INFO [_get_local_node] Local node name: srvnode-3.data.private

 Local node id: 3

2021-07-23 06:05:13 alert_monitor INFO [generate_iem] Sending an IEC: "IEC:IS0080010002:Server srvnode-1.data.private has joined back the cluster. System is restored. Extra Info: host=srvnode-1.data.private; status=member;" to syslog
2021-07-23 06:05:13 alert_monitor INFO [process_alert] Sent IEM alert from the node - name: srvnode-3.data.private
, id: 3

  </code>
</pre>

### CODE REVIEW

File: ha/alert/node_alert_monitor.py [hunk 1]
```diff
@@ -27,9 +30,53 @@ def __init__(self):
...
+        self.process = SimpleCommand()
+
+    def _get_online_nodes(self):
+        """
+        Get list of online nodes ids.
+        """
+        online_nodes_xml = self.process.run_cmd("crm_mon --as-xml")
+        # saving the xml file
+        with open('nodes.xml', 'w+') as f:
+            f.write(online_nodes_xml[0])
```
Comment (indrajitzagade) [inline ha/alert/node_alert_monitor.py:42]: Can't we get the work done without writing to a file?

File: ha/alert/node_alert_monitor.py [hunk 2]
```diff
@@ -27,9 +30,53 @@ def __init__(self):
         Init node alert monitor
         """
         super(NodeAlertMonitor, self).__init__()
+        self.process = SimpleCommand()
+
+    def _get_online_nodes(self):
+        """
+        Get list of online nodes ids.
+        """
+        online_nodes_xml = self.process.run_cmd("crm_mon --as-xml")
```
Comment (indrajitzagade) [inline ha/alert/node_alert_monitor.py:39]: Can we move all such string literals to const.py file in this directory?

File: ha/alert/node_alert_monitor.py (hunk 1 above)
Comment (shindeam) [reply ha/alert/node_alert_monitor.py:42]: No we can't, for ET.parse('nodes.xml') method it needs file hence written in that way.

File: ha/alert/node_alert_monitor.py (hunk 2 above)
Comment (shindeam) [reply ha/alert/node_alert_monitor.py]: Done

File: ha/alert/node_alert_monitor.py (hunk 1 above)
Comment (ajaysrivas) [reply ha/alert/node_alert_monitor.py:43]: @shindeam Please look at the code ha/core/controllers/pcs/cluster_status.py. You don't need to save output to a file.

File: ha/alert/const.py [hunk 3]
```diff
@@ -18,6 +18,10 @@
 
 from ha.const import _DELIM
 
+GET_ONLINE_NODES_CMD = "crm_mon --as-xml"
+GET_LOCAL_NODE_ID_CMD = "crm_node -i"
+GET_LOCAL_NODE_NAME_CMD = "crm_node -n"
+
```
Comment (ajay-paratmandali) [inline ha/alert/const.py:24]: GET_ONLINE_NODES_CMD = "crm_mon --as-xml"
GET_LOCAL_NODE_ID_CMD = "crm_node -i"
GET_LOCAL_NODE_NAME_CMD = "crm_node -n"

move them to
ha.const as generic api

Comment (shindeam) [reply ha/alert/const.py]: Done

File: ha/alert/node_alert_monitor.py (hunk 1 above)
Comment (shindeam) [reply ha/alert/node_alert_monitor.py]: Git it, taken code and updated our code by referencing it.

**Review by ajay-paratmandali — APPROVED** [review]

**Review by indrajitzagade — APPROVED** [review]

**Pull request merged**